# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsContainerFrame.cpp
# Commit: 5f30bde41790
# Full Hash: 5f30bde4179009fd405f8bd8c924b1f877874d42
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2019-08-15 19:33:18
# Regressor Bug: 1105868
# File Overlap Count: 2
# Description:
#   Bug 1574123 - Use nsLayoutUtils::GetMarkerFrame() more to avoid manual null-checks.  r=emilio
#   
#   (This basically addresses the review comments that I missed in
#   bug 1105868 part 4. My bad.)
#   
# ==============================================================================

diff -r 04f03c56a77f -r 5f30bde41790 layout/generic/nsContainerFrame.cpp
--- a/layout/generic/nsContainerFrame.cpp	Thu Aug 15 12:51:58 2019 +0000
+++ b/layout/generic/nsContainerFrame.cpp	Thu Aug 15 14:23:37 2019 +0000
@@ -1732,25 +1732,19 @@
 
 #ifdef ACCESSIBILITY
 void nsContainerFrame::GetSpokenMarkerText(nsAString& aText) const {
+  aText.Truncate();
   const nsStyleList* myList = StyleList();
   if (myList->GetListStyleImage()) {
     char16_t kDiscCharacter = 0x2022;
     aText.Assign(kDiscCharacter);
     aText.Append(' ');
-  } else {
-    nsIContent* markerPseudo = nsLayoutUtils::GetMarkerPseudo(GetContent());
-    if (nsIFrame* marker = markerPseudo->GetPrimaryFrame()) {
-      if (nsBulletFrame* bullet = do_QueryFrame(marker)) {
-        bullet->GetSpokenText(aText);
-      } else {
-        ErrorResult err;
-        markerPseudo->GetTextContent(aText, err);
-        if (err.Failed()) {
-          aText.Truncate();
-        }
-      }
+    return;
+  }
+  if (nsIFrame* marker = nsLayoutUtils::GetMarkerFrame(GetContent())) {
+    if (nsBulletFrame* bullet = do_QueryFrame(marker)) {
+      bullet->GetSpokenText(aText);
     } else {
-      aText.Truncate();
+      marker->GetContent()->GetTextContent(aText, IgnoreErrors());
     }
   }
 }
